lib/chibi/math/prime-test.sld

```diff
@@ -1,7 +1,22 @@
 (define-library (chibi math prime-test)
   (export run-tests)
-  (import (scheme base) (chibi math prime) (chibi test))
+  (import (scheme base) (chibi math prime) (chibi test) (srfi 27))
   (begin
+    (define (with-random-stupidity avoid thunk)
+      (let ((random-integer/actual random-integer))
+	(define (random-integer/unfortunate n)
+	  (let ((r (random-integer/actual n)))
+	    (if (avoid r)
+		(random-integer/unfortunate n)
+		r)))
+	;; TODO: this will not be thread safe
+	(dynamic-wind
+	    (lambda ()
+	      (set! random-integer random-integer/unfortunate))
```

Comment:
set!ing an imported variable is not legal R7RS.  I'm surprised I didn't make this an error.  It's ok for now, I'll fix the testing later.

Reply:
Yeah, I was in the middle of working out something complicated that respected hygiene and then I noticed `set!` just worked, which surprised me, too (Chez also forbids this).  I've been inactive since the R5RS era (... still kind of freaked out that they managed to agree on a module system).

And now I'm realizing that since my test case < 341550071728321 (where is this number from?) `random-integer` is not even being invoked at all, so I think I'll just remove this shit since it's evil.

Reply:
I can't recall where that number is from...

Reply:
Ah, it's from the observation that if n < 341,550,071,728,321, it is enough to test a = 2, 3, 5, 7, 11, 13, and 17 (http://www.math.dartmouth.edu/~carlp/PDF/paper25.pdf), although we're still doing extra work in those cases.